pilot/pkg/serviceregistry/aggregate/controller.go

```diff
@@ -233,7 +233,7 @@ func (c *Controller) GetProxyServiceInstances(node *model.Proxy) ([]*model.Servi
 			errs = multierror.Append(errs, err)
 		} else if len(instances) > 0 {
 			out = append(out, instances...)
-			node.ClusterID = r.Cluster()
+			node.ClusterID = instances[0].Endpoint.Locality.ClusterID
```

Comment:
This is always the same for all instances, right?

Reply:
I guess if we had multiple levels of aggregate controllers this would be wrong, but I don't think that's done in practice today. I believe the only controller that sets `Cluster` is k8s, which will not assign cluster ID to all endpoints properly.  So yes, they should be identical.  I don't love this code ... we should eventually get rid of `Cluster` from the controllers altogether and have it just be a property of endpoints.